File: backend/app/services/screening_config.py
# MedPulse screening configuration
# Questions are intentionally written in normal language.

SCREENING_QUESTIONS = {
    "general": [
        {
            "id": "age",
            "question": "How old are you?",
            "type": "number",
            "required": True
        },
        {
            "id": "gender",
            "question": "What is your gender?",
            "type": "choice",
            "options": ["Male", "Female"],
            "required": True
        }
    ],

    "hypertension": [
        {
            "id": "blood_pressure",
            "question": "Do you know your recent blood pressure reading?",
            "type": "blood_pressure",
            "required": True
        },
        {
            "id": "bmi",
            "question": "Do you know your height and weight?",
            "type": "height_weight",
            "required": False
        }
    ],

    "anemia": [
        {
            "id": "hemoglobin",
            "question": "Do you have a recent blood test showing your hemoglobin level?",
            "type": "number",
            "required": True
        },
        {
            "id": "tiredness",
            "question": "Have you been feeling unusually tired or weak?",
            "type": "yes_no",
            "required": False
        },
        {
            "id": "dizziness",
            "question": "Have you been feeling dizzy or light-headed?",
            "type": "yes_no",
            "required": False
        }
    ],

    "diabetes": [
        {
            "id": "frequent_urination",
            "question": "Have you been needing to urinate more often than usual?",
            "type": "yes_no",
            "required": False
        },
        {
            "id": "excessive_thirst",
            "question": "Have you been unusually thirsty?",
            "type": "yes_no",
            "required": False
        },
        {
            "id": "blood_glucose",
            "question": "Do you have a recent blood sugar reading?",
            "type": "number",
            "required": False
        }
    ],

    "kidney_disease": [
        {
            "id": "creatinine",
            "question": "Do you have a recent blood test showing your kidney function or creatinine level?",
            "type": "number",
            "required": False
        },
        {
            "id": "swelling",
            "question": "Have you noticed unusual swelling in your legs, feet, hands or face?",
            "type": "yes_no",
            "required": False
        },
        {
            "id": "urine_change",
            "question": "Have you noticed a change in how often you urinate or what your urine looks like?",
            "type": "yes_no",
            "required": False
        }
    ],

    "liver_disease": [
        {
            "id": "jaundice",
            "question": "Have your eyes or skin looked unusually yellow?",
            "type": "yes_no",
            "required": False
        },
        {
            "id": "abdominal_pain",
            "question": "Have you had ongoing pain or discomfort in the upper-right part of your stomach?",
            "type": "yes_no",
            "required": False
        }
    ]
}

OPTIONAL_WARNING = (
    "You can continue without optional information, but providing it "
    "can improve screening accuracy."
)

File: frontend/src/screeningConfig.ts
export type ScreeningField = {
  key: string;
  label: string;
  type: "profile" | "symptom" | "measurement" | "report";
  question: string;
  help?: string;
};

export const SCREENING_CONFIG: Record<string, ScreeningField[]> = {
  diabetes: [
    {
      key: "age",
      label: "Age",
      type: "profile",
      question: "How old are you?"
    },
    {
      key: "bmi",
      label: "Body size",
      type: "profile",
      question: "We'll calculate this automatically from your height and weight."
    },
    {
      key: "glucose",
      label: "Blood sugar",
      type: "report",
      question: "Do you have a recent blood-test report?",
      help: "You don't need to know your blood-sugar number. Upload the report and we'll look for the relevant information."
    },
    {
      key: "blood_pressure",
      label: "Blood pressure",
      type: "measurement",
      question: "Do you know your recent blood-pressure reading?",
      help: "For example, 120/80."
    }
  ],

  heart_disease: [
    {
      key: "age",
      label: "Age",
      type: "profile",
      question: "How old are you?"
    },
    {
      key: "chest_discomfort",
      label: "Chest discomfort",
      type: "symptom",
      question: "Have you experienced chest discomfort or pressure?",
      help: "You don't need to know the medical term for it."
    },
    {
      key: "shortness_of_breath",
      label: "Breathing",
      type: "symptom",
      question: "Do you get short of breath more easily than usual?"
    },
    {
      key: "heart_report",
      label: "Heart report",
      type: "report",
      question: "Do you have a recent ECG or heart-test report?",
      help: "Upload it if you have one. You don't need to understand the medical numbers."
    }
  ],

  hypertension: [
    {
      key: "age",
      label: "Age",
      type: "profile",
      question: "How old are you?"
    },
    {
      key: "bmi",
      label: "Body size",
      type: "profile",
      question: "We'll calculate this automatically from your height and weight."
    },
    {
      key: "blood_pressure",
      label: "Blood pressure",
      type: "measurement",
      question: "Do you know your recent blood-pressure reading?",
      help: "For example, 120/80. If you don't know it, you can use a recent medical report."
    },
    {
      key: "blood_pressure_report",
      label: "Blood-pressure report",
      type: "report",
      question: "Don't know your blood-pressure numbers?",
      help: "Upload a recent medical report instead."
    }
  ],

  anemia: [
    {
      key: "age",
      label: "Age",
      type: "profile",
      question: "How old are you?"
    },
    {
      key: "blood_test_report",
      label: "Blood test",
      type: "report",
      question: "Do you have a recent blood-test report?",
      help: "MedPulse can use the relevant blood information. You don't need to know terms like MCH, MCHC or MCV."
    }
  ]
};
